### PR TITLE
Signals: what do they mean for _you_, specifically?

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
super wip, not worth reading yet.

the words here are not representative of anything atm

Notes:

# author TODOS / notes

- do less name dropping of projects
- mention relays as a useful pattern
- focus on the goal: for each of the three audiences, provide high level background, entice excitment about all the practical benefits of spec-signals, and provide calls to action / to try out integration / playing with the polyfill, etc
- GOAL: help people grasp the implications of the proposal: https://twitter.com/bobaekang/status/1777479259403047204
- make the text in the timeline chart bigger: https://github.com/NullVoxPopuli/website/pull/146/files#diff-43616e7ee8fb5323b7a16603833d084ee07bf8124f23319771da605d265f24fb

## Extract for separate blog post

- remove content on sync effects 
- effects less in general, because they aren't well defined (too many kinds) 
- talk about pull-based and glitch-free behavior, emergent patterns
- example of something in the ecosystem that would benefit from being universal

### CODE REVIEW

**Review by shaylew — COMMENTED** [review]
Leaving some specific comments, but my general comment is "yay, blog post!"

File: content/2024-04-24-signals.md [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+
+There are two places to watch for developments there:
+- [HTML Template Instantiation](https://github.com/WICG/webcomponents/blob/gh-pages/proposals/Template-Instantiation.md)
+- [DOM Parts](https://github.com/WICG/webcomponents/pull/1023)
+
+## What are the requirements?
+
+- [Autotracking](https://github.com/proposal-signals/proposal-signals/issues/161) - the main feature of the Signals proposal that makes it more than just values and functions.
+- Memory allocations should be kept to a minimum.
+- `Signal.State` should be sub-classable so that frameworks can add their own functionality on top.
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply content/2024-04-24-signals.md:186]: yeah, idk -- I'm going to join the call this tuesday, -- so i'll try to ask the group then

[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply content/2024-04-24-signals.md:186]: from @littledan
- goal: make things extensible with fewer object allocations -- and using fields and subclasses are one such way to do that 
-

File: content/2024-04-24-signals.md [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+| root state | a value representing state, it cannot depend on anything within the graph as it is a _root_ |
+| atom | an alias of root state |
+| derived data | a derivation on or of state(s), which may or may not be cached, in a pure form this could be written as `derived = f(state)` 
+| computed | alias of derived, tho this has had different meanings throughout the years in different ecosystems. In this context, specific to [the proposal][signal-proposal], it is a cached derived value.
+| formula | alias of derived, part of the spreadsheet analogy
+| upstream | the node that is "upstream of" a given node is closer to the root.
+| downstream | the node that is "downstream of" a given node is further away from the root.
+| sink | a consumer of sources or values
+| source | root state, the source of a value
+| auto-tracking | the process of automatically tracking if a computed or derived value should recompute next time it is read |
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply content/2024-04-24-signals.md:123]: thank you!!

File: content/2024-04-24-signals.md [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+### Some helpful terminology and definitions that various folks use
+
+As people talk about Signals, auto-tracking, Runes, or other forms of reactivity, it may be heplful to have everything defined / so we can all be on the same page / reduce any potential ambiguity or uncertainty. Not all of these will be used here, in this post, but... more _for your (and my own) information[^terms]_.
+
+| term | meaning |
+| ---- | ------- |
+| node | part of the reactive graph |
+| value | an alias of node |
+| cell | could be an alias of node, but is often root state, part of the spreadsheet analogy |
+| root state | a value representing state, it cannot depend on anything within the graph as it is a _root_ |
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply content/2024-04-24-signals.md:114]: than you!!

File: content/2024-04-24-signals.md [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+
+## What even are Signals?
+
+Signals are a means to enable efficient computation of state for a particular viewer. The viewer in many cases is the DOM, but could extend to any observable[^observable] boundary, a WebSocket, Database Connection, etc.
+
+
+The exact way in which this works isn't important right now (and also the exact details will probably change over time as the proposal is still in [Stage 0](https://tc39.es/process-document/)) -- perhaps a topic for another time!
+
+
+### Some helpful terminology and definitions that various folks use
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply content/2024-04-24-signals.md:105]: this would be cool to see!

**Pull request merged**